pkg/core/object/fmt.go

```diff
@@ -199,9 +199,13 @@ func (v *FormatValidator) validateSignatureKey(obj *object.Object) error {
 		return errors.New("session token is not for object's signer")
 	}
 
-	if _, err := icrypto.VerifyTokenSignature(token); err != nil {
+	tokenSigner, err := icrypto.VerifyTokenSignature(token)
+	if err != nil {
 		return fmt.Errorf("verify session token signature: %w", err)
 	}
+	if tokenSigner != token.Issuer() {
```

Comment:
Better just inline this check, it's always needed for both types of tokens.

Reply:
done